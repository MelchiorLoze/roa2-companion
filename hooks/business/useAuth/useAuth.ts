import { useEffect } from 'react';

import { useSession } from '@/contexts';
import { useGetEntityToken, useLoginWithEmail } from '@/hooks/data';

export const useAuth = () => {
  const { isValid, shouldRenew, setSession, isLoading: isSessionLoading } = useSession();
  const { session, loginWithEmail, isLoading: isLoginLoading, isError } = useLoginWithEmail();
  const { newSession, renew } = useGetEntityToken();

  const logout = () => setSession(null);

  useEffect(() => {
    if (session) setSession(session);
  }, [session, setSession]);

  useEffect(() => {
    if (newSession) setSession(newSession);
  }, [newSession, setSession]);

  useEffect(() => {
    if (shouldRenew) renew();
  }, [shouldRenew, renew]);

  return {
    isLoggedIn: isValid,
    login: loginWithEmail,
    logout,
    isLoading: isSessionLoading || isLoginLoading,
    isError,
  };
};
