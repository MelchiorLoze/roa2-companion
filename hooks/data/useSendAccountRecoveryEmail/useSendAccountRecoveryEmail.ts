import { useMutation } from '@tanstack/react-query';

import { TITLE_ID } from '@/constants';
import { useHttpClient } from '@/hooks/core';

export const useSendAccountRecoveryEmail = () => {
  const httpClient = useHttpClient();

  const { mutate, isPending, isError } = useMutation({
    mutationFn: (email: string) =>
      httpClient.post<void>('/Client/SendAccountRecoveryEmail', {
        TitleId: TITLE_ID,
        Email: email,
      }),
  });

  return {
    sendRecoveryEmail: mutate,
    isLoading: isPending,
    isError,
  };
};
