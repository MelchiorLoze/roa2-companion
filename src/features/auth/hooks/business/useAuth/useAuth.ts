import { useEffect } from 'react';

import { useSession } from '../../../contexts/SessionContext/SessionContext';
import { useLoginWithEmail } from '../../data/useLoginWithEmail/useLoginWithEmail';

export const useAuth = () => {
  const { isValid, setSession, clearSession, isLoading: isLoadingSession } = useSession();
  const { session, loginWithEmail, isLoading: isLoadingLogin, isError } = useLoginWithEmail();

  useEffect(() => {
    if (session) setSession(session);
  }, [session, setSession]);

  return {
    isLoggedIn: isValid,
    login: loginWithEmail,
    logout: clearSession,
    isLoading: isLoadingSession || isLoadingLogin,
    isError,
  } as const;
};
