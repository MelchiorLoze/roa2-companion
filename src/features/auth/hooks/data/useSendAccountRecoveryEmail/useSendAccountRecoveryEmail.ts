import { useMutation } from '@tanstack/react-query';

import { TITLE_ID } from '@/constants';
import { useGameApiClient } from '@/hooks/apiClients/useGameApiClient/useGameApiClient';

export const useSendAccountRecoveryEmail = () => {
  const apiClient = useGameApiClient();

  const { mutate, isPending, isSuccess, isError } = useMutation({
    mutationFn: (email: string) =>
      apiClient.post<void>('/Client/SendAccountRecoveryEmail', {
        body: {
          TitleId: TITLE_ID,
          Email: email,
        },
      }),
  });

  return {
    sendRecoveryEmail: mutate,
    isLoading: isPending,
    isSuccess,
    isError,
  } as const;
};
