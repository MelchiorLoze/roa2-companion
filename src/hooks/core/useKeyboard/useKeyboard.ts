import { useEffect, useState } from 'react';
import { Keyboard } from 'react-native';

export const useKeyboard = () => {
  const [isVisible, setIsVisible] = useState(false);

  useEffect(() => {
    const showSubscription = Keyboard.addListener('keyboardDidShow', () => setIsVisible(true));
    const hideSubscription = Keyboard.addListener('keyboardDidHide', () => setIsVisible(false));

    return () => {
      showSubscription.remove();
      hideSubscription.remove();
    };
  }, []);

  return { isKeyboardVisible: isVisible };
};
