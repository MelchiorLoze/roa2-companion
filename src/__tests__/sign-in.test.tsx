import { fireEvent, render, screen, within } from '@testing-library/react-native';
import { Redirect } from 'expo-router';

import SignIn from '@/app/sign-in';
import { useAuth } from '@/features/auth/hooks/business/useAuth/useAuth';
import { useSendAccountRecoveryEmail } from '@/features/auth/hooks/data/useSendAccountRecoveryEmail/useSendAccountRecoveryEmail';
import { useKeyboard } from '@/hooks/core/useKeyboard/useKeyboard';

jest.mock('expo-router');
const RedirectMock = jest.mocked(Redirect);

jest.mock('@/hooks/core/useKeyboard/useKeyboard');
const useKeyboardMock = jest.mocked(useKeyboard);

jest.mock('@/features/auth/hooks/data/useSendAccountRecoveryEmail/useSendAccountRecoveryEmail');
const useSendAccountRecoveryEmailMock = jest.mocked(useSendAccountRecoveryEmail);
const defaultSendRecoveryEmailState: ReturnType<typeof useSendAccountRecoveryEmail> = {
  sendRecoveryEmail: jest.fn(),
  isLoading: false,
  isSuccess: false,
  isError: false,
};

jest.mock('@/features/auth/hooks/business/useAuth/useAuth');
const useAuthMock = jest.mocked(useAuth);
const defaultAuthState: ReturnType<typeof useAuth> = {
  isLoggedIn: false,
  login: jest.fn(),
  logout: jest.fn(),
  isLoading: false,
  isError: false,
};

const screenTitle = 'Login to your in-game account';

const renderComponent = () => {
  const result = render(<SignIn />);

  expect(useAuthMock).toHaveBeenCalledTimes(1);
  expect(defaultAuthState.login).not.toHaveBeenCalled();

  return result;
};

describe('SignIn', () => {
  beforeEach(() => {
    useKeyboardMock.mockReturnValue({ isKeyboardVisible: false });
    useAuthMock.mockReturnValue(defaultAuthState);
    useSendAccountRecoveryEmailMock.mockReturnValue(defaultSendRecoveryEmailState);
  });

  it('matches the snapshot', () => {
    const tree = renderComponent().toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    renderComponent();

    screen.getByTestId('disclaimer');
    screen.getByText(screenTitle);

    const emailInput = screen.getByPlaceholderText('EMAIL');
    expect(emailInput).toHaveDisplayValue('');
    const passwordInput = screen.getByPlaceholderText('PASSWORD');
    expect(passwordInput).toHaveDisplayValue('');

    expect(screen.queryByText('Invalid email or password')).toBeNull();
    screen.getByRole('button', { name: 'Forgot your password?' });

    screen.getByRole('button', { name: 'Login' });
  });

  it('renders loading state', () => {
    useAuthMock.mockReturnValue({
      ...defaultAuthState,
      isLoading: true,
    });

    renderComponent();

    screen.getByTestId('spinner');
    expect(screen.queryByText(screenTitle)).toBeNull();
  });

  it('redirects to store when already logged in', () => {
    useAuthMock.mockReturnValue({
      ...defaultAuthState,
      isLoggedIn: true,
    });

    renderComponent();

    expect(RedirectMock).toHaveBeenCalledTimes(1);
    expect(RedirectMock).toHaveBeenCalledWith({ href: '/store' }, undefined);
    expect(screen.queryByText(screenTitle)).toBeNull();
  });

  it('shows an error message when submitting with empty email or password', () => {
    renderComponent();

    const loginButton = screen.getByRole('button', { name: 'Login' });
    fireEvent.press(loginButton);

    screen.getByText('Invalid email or password');
    expect(defaultAuthState.login).not.toHaveBeenCalled();
  });

  it('shows error message when login fails', () => {
    useAuthMock.mockReturnValue({
      ...defaultAuthState,
      isError: true,
    });

    renderComponent();

    screen.getByText('Invalid email or password');
  });

  it('calls login function with email and password', () => {
    renderComponent();

    const emailInput = screen.getByPlaceholderText('EMAIL');
    const passwordInput = screen.getByPlaceholderText('PASSWORD');
    const loginButton = screen.getByRole('button', { name: 'Login' });

    fireEvent.changeText(emailInput, 'kragg@example.com');
    expect(emailInput).toHaveDisplayValue('kragg@example.com');
    fireEvent.changeText(passwordInput, 'r0ck');
    expect(passwordInput).toHaveDisplayValue('r0ck');
    fireEvent.press(loginButton);

    expect(defaultAuthState.login).toHaveBeenCalledTimes(1);
    expect(defaultAuthState.login).toHaveBeenCalledWith({
      email: 'kragg@example.com',
      password: 'r0ck',
    });
  });

  it('hides error message when calling login with valid credentials', () => {
    renderComponent();

    const emailInput = screen.getByPlaceholderText('EMAIL');
    const passwordInput = screen.getByPlaceholderText('PASSWORD');
    const loginButton = screen.getByRole('button', { name: 'Login' });

    fireEvent.press(loginButton);
    screen.getByText('Invalid email or password');

    fireEvent.changeText(emailInput, 'clairen@example.com');
    fireEvent.changeText(passwordInput, '1MM4TUR3');
    fireEvent.press(loginButton);

    expect(screen.queryByText('Invalid email or password')).toBeNull();
  });

  it('shows reset password dialog when clicking on forgot password', () => {
    renderComponent();

    const forgotPasswordButton = screen.getByRole('button', { name: 'Forgot your password?' });
    fireEvent.press(forgotPasswordButton);
    const withinDialog = within(screen.getByTestId('dialog'));

    withinDialog.getByText(
      'After submitting, you will receive an email from Aether Studios allowing you to reset your password',
    );
  });

  it('fills email field when reset password dialog is closed with email', () => {
    const { rerender } = renderComponent();

    const emailInput = screen.getByPlaceholderText('EMAIL');
    expect(emailInput).toHaveDisplayValue('');

    const forgotPasswordButton = screen.getByRole('button', { name: 'Forgot your password?' });
    fireEvent.press(forgotPasswordButton);
    const withinDialog = within(screen.getByTestId('dialog'));

    const resetPasswordEmailInput = withinDialog.getByPlaceholderText('EMAIL');
    fireEvent.changeText(resetPasswordEmailInput, 'kragg@example.com');

    const resetPasswordButton = withinDialog.getByRole('button', { name: 'Reset password' });
    fireEvent.press(resetPasswordButton);

    useSendAccountRecoveryEmailMock.mockReturnValue({
      ...defaultSendRecoveryEmailState,
      isSuccess: true,
    });
    rerender(<SignIn />);

    const closeButton = withinDialog.getByRole('button', { name: 'Ok' });
    fireEvent.press(closeButton);

    expect(emailInput).toHaveDisplayValue('kragg@example.com');
  });

  it('hides the disclaimer when the keyboard is visible', () => {
    useKeyboardMock.mockReturnValue({ isKeyboardVisible: true });

    renderComponent();

    expect(screen.queryByTestId('disclaimer')).toBeNull();
  });
});
