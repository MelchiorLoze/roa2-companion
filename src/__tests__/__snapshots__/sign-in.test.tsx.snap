// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn matches the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "gap": 24,
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 24,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontFamily": "FranklinGothicDemiCond-Italic",
        "fontSize": 24,
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
  >
    Login to your
    

    in-game account
  </Text>
  <View
    style={
      {
        "gap": 12,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "gap": 8,
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        autoComplete="email"
        cursorColor="white"
        onChangeText={[Function]}
        placeholder="EMAIL"
        placeholderTextColor="#676767"
        style={
          {
            "backgroundColor": "#161616",
            "borderColor": "#FFDF77",
            "borderWidth": 1,
            "color": "white",
            "height": 48,
            "padding": 8,
            "width": "100%",
          }
        }
        value=""
      />
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        {
          "gap": 8,
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        autoComplete="current-password"
        cursorColor="white"
        onChangeText={[Function]}
        placeholder="PASSWORD"
        placeholderTextColor="#676767"
        secureTextEntry={true}
        style={
          {
            "backgroundColor": "#161616",
            "borderColor": "#FFDF77",
            "borderWidth": 1,
            "color": "white",
            "height": 48,
            "padding": 8,
            "width": "100%",
          }
        }
        value=""
      />
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          role="button"
          style={
            {
              "flex": 1,
              "padding": 4,
            }
          }
        >
          <Text
            style={
              {
                "color": "#CACEFF",
                "fontFamily": "FranklinGothicDemiCond-Italic",
                "fontSize": 14,
                "textAlign": "right",
                "textDecorationLine": "underline",
              }
            }
          >
            Forgot your password?
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="button"
  >
    <ViewManagerAdapter_ExpoLinearGradient
      colors={
        [
          4281146742,
          4284183501,
        ]
      }
      endPoint={
        [
          1,
          0,
        ]
      }
      startPoint={
        [
          0,
          0,
        ]
      }
      style={
        {
          "padding": 4,
          "paddingHorizontal": 24,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "FranklinGothicDemiCond-Regular",
              "fontSize": 16,
              "textAlign": "center",
              "textTransform": "uppercase",
            },
            false,
          ]
        }
      >
        Login
      </Text>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "flexDirection": "row",
          "gap": 8,
        },
        {
          "bottom": 24,
          "position": "absolute",
        },
      ]
    }
    testID="disclaimer"
  >
    <
      name="information-circle-sharp"
      style={
        {
          "color": "#C8C8D5",
          "fontSize": 24,
        }
      }
    />
    <Text
      style={
        {
          "color": "#C8C8D5",
          "flexShrink": 1,
          "fontFamily": "FranklinGothicDemiCond-Italic",
          "fontSize": 14,
        }
      }
    >
      This project is a fan-made creation and is not affiliated with Aether Studios or Offbrand Games in any official capacity. The app works directly between your device and the official game servers, with no third parties involved in transferring or processing your data. 
      <Text
        style={
          {
            "color": "white",
            "textDecorationLine": "underline",
          }
        }
      >
        Learn more...
      </Text>
    </Text>
  </View>
</View>
`;

exports[`SignIn matches the snapshot 2`] = `
[
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "gap": 24,
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 24,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontFamily": "FranklinGothicDemiCond-Italic",
          "fontSize": 24,
          "textAlign": "center",
          "textTransform": "uppercase",
          "width": "100%",
        }
      }
    >
      Login to your
      

      in-game account
    </Text>
    <View
      style={
        {
          "gap": 12,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "gap": 8,
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          autoComplete="email"
          cursorColor="white"
          onChangeText={[Function]}
          placeholder="EMAIL"
          placeholderTextColor="#676767"
          style={
            {
              "backgroundColor": "#161616",
              "borderColor": "#FFDF77",
              "borderWidth": 1,
              "color": "white",
              "height": 48,
              "padding": 8,
              "width": "100%",
            }
          }
          value=""
        />
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        />
      </View>
      <View
        style={
          {
            "gap": 8,
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          autoComplete="current-password"
          cursorColor="white"
          onChangeText={[Function]}
          placeholder="PASSWORD"
          placeholderTextColor="#676767"
          secureTextEntry={true}
          style={
            {
              "backgroundColor": "#161616",
              "borderColor": "#FFDF77",
              "borderWidth": 1,
              "color": "white",
              "height": 48,
              "padding": 8,
              "width": "100%",
            }
          }
          value=""
        />
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
            style={
              {
                "flex": 1,
                "padding": 4,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#CACEFF",
                  "fontFamily": "FranklinGothicDemiCond-Italic",
                  "fontSize": 14,
                  "textAlign": "right",
                  "textDecorationLine": "underline",
                }
              }
            >
              Forgot your password?
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
    >
      <ViewManagerAdapter_ExpoLinearGradient
        colors={
          [
            4281146742,
            4284183501,
          ]
        }
        endPoint={
          [
            1,
            0,
          ]
        }
        startPoint={
          [
            0,
            0,
          ]
        }
        style={
          {
            "padding": 4,
            "paddingHorizontal": 24,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "white",
                "fontFamily": "FranklinGothicDemiCond-Regular",
                "fontSize": 16,
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              false,
            ]
          }
        >
          Login
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "flexDirection": "row",
            "gap": 8,
          },
          {
            "bottom": 24,
            "position": "absolute",
          },
        ]
      }
      testID="disclaimer"
    >
      <
        name="information-circle-sharp"
        style={
          {
            "color": "#C8C8D5",
            "fontSize": 24,
          }
        }
      />
      <Text
        style={
          {
            "color": "#C8C8D5",
            "flexShrink": 1,
            "fontFamily": "FranklinGothicDemiCond-Italic",
            "fontSize": 14,
          }
        }
      >
        This project is a fan-made creation and is not affiliated with Aether Studios or Offbrand Games in any official capacity. The app works directly between your device and the official game servers, with no third parties involved in transferring or processing your data. 
        <Text
          style={
            {
              "color": "white",
              "textDecorationLine": "underline",
            }
          }
        >
          Learn more...
        </Text>
      </Text>
    </View>
  </View>,
  <View>
    <Modal
      animationType="fade"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      statusBarTranslucent={true}
      transparent={true}
      visible={true}
    >
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#161616AA",
              "height": "100%",
              "position": "absolute",
              "width": "100%",
            }
          }
          testID="overlay"
        />
        <View
          style={
            {
              "backgroundColor": "#0E0B2A",
              "borderColor": "#FFDF77",
              "borderWidth": 2,
              "gap": 16,
              "padding": 16,
              "width": "85%",
            }
          }
          testID="dialog"
        >
          <Text
            style={
              {
                "color": "white",
                "fontFamily": "FranklinGothicDemiCond-Regular",
                "fontSize": 16,
              }
            }
          >
            After submitting, you will receive an email from Aether Studios allowing you to reset your password
          </Text>
          <View
            style={
              {
                "gap": 8,
              }
            }
          >
            <TextInput
              autoCapitalize="none"
              autoComplete="email"
              cursorColor="white"
              onChangeText={[Function]}
              placeholder="EMAIL"
              placeholderTextColor="#676767"
              style={
                {
                  "backgroundColor": "#161616",
                  "borderColor": "#FFDF77",
                  "borderWidth": 1,
                  "color": "white",
                  "height": 48,
                  "padding": 8,
                  "width": "100%",
                }
              }
              value=""
            />
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
          >
            <ViewManagerAdapter_ExpoLinearGradient
              colors={
                [
                  4281146742,
                  4284183501,
                ]
              }
              endPoint={
                [
                  1,
                  0,
                ]
              }
              startPoint={
                [
                  0,
                  0,
                ]
              }
              style={
                {
                  "padding": 4,
                  "paddingHorizontal": 24,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "white",
                      "fontFamily": "FranklinGothicDemiCond-Regular",
                      "fontSize": 16,
                      "textAlign": "center",
                      "textTransform": "uppercase",
                    },
                    false,
                  ]
                }
              >
                Reset password
              </Text>
            </ViewManagerAdapter_ExpoLinearGradient>
          </View>
        </View>
      </View>
    </Modal>
  </View>,
]
`;
